base/task/sequence_manager/task_queue_impl.cc

```diff
@@ -1402,6 +1402,9 @@ void TaskQueueImpl::ActivateDelayedFenceIfNeeded(const Task& task) {
   main_thread_only().delayed_fence = absl::nullopt;
 }
 
+#if !defined(STARBOARD)
```

Comment:
Why does this have an impact ? All of the code here seems to be taking an early out if tracing is not enabled - do we somehow enable tracing ? 

Or is the macro overhead of TRACE_EVENT_CATEGORY_GROUP_ENABLED so high ? If that's the case, maybe it's better to force the macro to be a no-op or false in gold/qa builds - as it's likely called elsewhere too ?

Reply:
In `//base/trace_event/base_tracing.h` there's a `#if BUILDFLAG(ENABLE_BASE_TRACING)` , when set to false it uses `#include "base/trace_event/trace_event_stub.h"` which should all reduce to compile-time no-op. 

Do we have `ENABLE_BASE_TRACING` on in Gold ?

Reply:
Disabling `ENABLE_BASE_TRACING` does not make `TRACE_EVENT_CATEGORY_GROUP_ENABLED` a no-op. We don't look at gold builds. This adds code to a very performance sensitive part, it showed up in the trace on a qa build.

Reply:
After doing a separate set of benchmark with just changing tracing (#4509), there appears to be no measurable improvement from this. I'll drop this change for now.